onnxruntime/test/util/include/test_dynamic_plugin_ep.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include "onnxruntime_cxx_api.h"
+
+#include "core/common/status.h"
+
+namespace onnxruntime {
+struct IExecutionProviderFactory;
+class IExecutionProvider;
+
+namespace logging {
+class Logger;
+}  // namespace logging
+
+namespace test {
+
+// `onnxruntime::test::dynamic_plugin_ep_infra` contains functions and types related to dynamically loaded plugin EP
+// unit testing infrastructure.
+namespace dynamic_plugin_ep_infra {
+
+// Note: `Initialize()` and `Shutdown()` are not thread-safe.
+// They should be called before and after calls to most of the other functions in this namespace.
+// The exception to this is `ParseInitializationConfig()`, which may be called before `Initialize()`.
+
+// Configuration for initializing the dynamic plugin EP infrastructure.
+struct InitializationConfig {
+  std::string ep_library_registration_name{};
+  std::string ep_library_path{};
+  std::vector<size_t> selected_ep_device_indices{0};
+  std::map<std::string, std::string> default_ep_options{};
+};
+
+// Parses `InitializationConfig` from JSON.
+// The configuration JSON object should have keys and values that match the `InitializationConfig` fields.
+// E.g.:
+// {
+//   "ep_library_registration_name": "example_plugin_ep",
+//   "ep_library_path": "/path/to/example_plugin_ep.dll",
+//   "selected_ep_device_indices": [1],
```

Comment:
Maybe this is a naive question, but why do we specify the indices of the OrtEpDevices instead of the EP names?

I would imagine that getting the indices would require the user to call GetEpDevices to get the indices.

Reply:
it's a naive implementation :). open to better ways of specifying it. yes, at the moment the user should check the EP devices first. this can be done with `onnxruntime_perf_test --list_ep_devices`.

specifying the particular EP devices seemed to be the most flexible, but perhaps it is just passing more complexity to the user. if we specify an EP name instead, what would be a sensible way to handle multiple EP devices associated with one EP name?